setup/src/main.ts

```diff
@@ -56,4 +80,95 @@ async function getSprklPrefixOrFail(): Promise<string> {
     } else {
         throw new Error(myError);
     }
-}
\ No newline at end of file
+} 
+
+/**
+    Return map of env vars to set depending on the workflow event(push, pull request or others).
+ */
+export async function autoRecipe(eventName: string): Promise<Map<string, string| number>> {
+    // get the workflow json which include all the data about the workflow
+    const workflowContext = JSON.parse(JSON.stringify(github.context.payload, undefined, 2));
+    if (eventName === 'push') {
+        return getPushEnvVarsOrFail(workflowContext);
+    } else if (eventName === 'pull_request') {
+        return await getPullRequestEnvVarsOrFail(workflowContext);
+    } else {
+        // return 'recent' recipe with last 10 commits
+        let envVarsMap = new Map<string, string|number>();
+        envVarsMap.set('SPRKL_RECIPE', 'recent');
+        envVarsMap.set('SPRKL_RECIPE_ATTRIBUTES_AMOUNT', 10);
```

Comment:
test this

Reply:
I tested the CLI with the env vars and it worked well